File: implementations/javascript/examples/lsf-to-json-example.ts
/**
 * Example demonstrating LSF to JSON conversion in TypeScript
 */

import { LSFEncoder, lsfToJson, lsfToJsonPretty } from '../src';

// Create some LSF data using the encoder
const encoder = new LSFEncoder();
const lsfString = encoder
    .startObject("user")
    .addField("id", 123)
    .addField("name", "John Doe")
    .addList("tags", ["admin", "developer", "tester"])
    .addTypedField("active", true, "bool")
    .addTypedField("salary", 75000.50, "float")
    .addTypedField("manager", null, "null")
    .startObject("company")
    .addField("name", "Acme Corp")
    .addField("industry", "Technology")
    .addTypedField("founded", 1985, "int")
    .toString();

console.log("LSF String (not human readable):");
console.log(lsfString);
console.log("\n" + "-".repeat(50) + "\n");

// Convert to compact JSON
const jsonString = lsfToJson(lsfString);
console.log("Compact JSON:");
console.log(jsonString);
console.log("\n" + "-".repeat(50) + "\n");

// Convert to pretty-printed JSON
const prettyJson = lsfToJsonPretty(lsfString);
console.log("Pretty JSON:");
console.log(prettyJson);
console.log("\n" + "-".repeat(50) + "\n");

// Custom indentation
const customJson = lsfToJson(lsfString, 4);
console.log("Custom Indentation (4 spaces):");
console.log(customJson);
console.log("\n" + "-".repeat(50) + "\n");

// Using a replacer function to filter out sensitive information
const replacer = (key: string, value: any) => {
    // Remove salary information for privacy
    if (key === 'salary') return '(redacted)';
    return value;
};

const filteredJson = lsfToJson(lsfString, 2, replacer);
console.log("With Sensitive Information Filtered:");
console.log(filteredJson); 

File: implementations/python/examples/lsf_to_json_example.py
"""
Example demonstrating LSF to JSON conversion in Python
"""

from lsf import LSFEncoder, lsf_to_json, lsf_to_json_pretty

# Create some LSF data using the encoder
encoder = LSFEncoder()
lsf_string = (encoder
    .start_object("user")
    .add_field("id", 123)
    .add_field("name", "John Doe")
    .add_list("tags", ["admin", "developer", "tester"])
    .add_typed_field("active", True, "bool")
    .add_typed_field("salary", 75000.50, "float")
    .add_typed_field("manager", None, "null")
    .start_object("company")
    .add_field("name", "Acme Corp")
    .add_field("industry", "Technology")
    .add_typed_field("founded", 1985, "int")
    .to_string())

print("LSF String (not human readable):")
print(lsf_string)
print("\n" + "-" * 50 + "\n")

# Convert to compact JSON
json_string = lsf_to_json(lsf_string)
print("Compact JSON:")
print(json_string)
print("\n" + "-" * 50 + "\n")

# Convert to pretty-printed JSON
pretty_json = lsf_to_json_pretty(lsf_string)
print("Pretty JSON:")
print(pretty_json)
print("\n" + "-" * 50 + "\n")

# Custom indentation
custom_json = lsf_to_json(lsf_string, indent=4)
print("Custom Indentation (4 spaces):")
print(custom_json)
print("\n" + "-" * 50 + "\n")

# Sorted keys
sorted_json = lsf_to_json(lsf_string, indent=2, sort_keys=True)
print("With Sorted Keys:")
print(sorted_json) 